e2e/test/scenarios/visualizations/reproductions/30057-30058-map-crash.cy.spec.js

```diff
@@ -0,0 +1,142 @@
+import {
+  popover,
+  restore,
+  visitQuestionAdhoc,
+  visualize,
+} from "e2e/support/helpers";
+
+import { SAMPLE_DB_ID } from "e2e/support/cypress_data";
+import { SAMPLE_DATABASE } from "e2e/support/cypress_sample_database";
+
+const { PEOPLE, PEOPLE_ID } = SAMPLE_DATABASE;
+
+const testQuery = {
+  type: "query",
+  query: {
+    "source-query": {
+      "source-table": PEOPLE_ID,
+      aggregation: [["count"]],
+      breakout: [
+        [
+          "field",
+          PEOPLE.LATITUDE,
+          { "base-type": "type/Float", binning: { strategy: "default" } },
+        ],
+        [
+          "field",
+          PEOPLE.LONGITUDE,
+          { "base-type": "type/Float", binning: { strategy: "default" } },
+        ],
+      ],
+    },
+  },
+  database: SAMPLE_DB_ID,
+};
+
+describe("issue 30057", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsNormalUser();
+  });
+
+  it("should still display visualization as a map after adding a filter (metabase#30057)", () => {
+    visitQuestionAdhoc({ dataset_query: testQuery }, { mode: "notebook" });
+
+    visualize();
+    addCountGreaterThan2Filter();
+    visualize();
+
+    cy.findByTestId("TableInteractive-root").should("not.exist");
+    cy.get(".PinMap").should("exist");
+  });
+
+  it("should still display visualization as a map after adding another column to group by", () => {
+    visitQuestionAdhoc({ dataset_query: testQuery }, { mode: "notebook" });
+
+    visualize();
+    groupByBirthDateColumn();
+    visualize();
+
+    cy.findByTestId("TableInteractive-root").should("not.exist");
+    cy.get(".PinMap").should("exist");
+  });
+
+  it("should still display visualization as a map after adding another aggregation", () => {
+    visitQuestionAdhoc({ dataset_query: testQuery }, { mode: "notebook" });
+
+    visualize();
+    addAverageOfLongitudeAggregation();
+    visualize();
+
+    cy.findByTestId("TableInteractive-root").should("not.exist");
+    cy.get(".PinMap").should("exist");
+  });
+
+  it("should change display to default after removing a column to group by when map is not sensible anymore", () => {
+    visitQuestionAdhoc({ dataset_query: testQuery }, { mode: "notebook" });
+
+    visualize();
+    removeLatitudeColumn();
+    visualize();
+
+    cy.get(".PinMap").should("not.exist");
+    cy.get(".LineAreaBarChart").should("exist");
+  });
+});
+
+describe("issue 30058", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsNormalUser();
+  });
+
+  it("should not crash visualization after adding a filter (metabase#30058)", () => {
+    visitQuestionAdhoc({
+      dataset_query: testQuery,
+      display: "map",
+      displayIsLocked: true,
+    });
+
+    addCountGreaterThan2Filter();
+    visualize();
+
+    cy.get(".Icon-warning").should("not.exist");
+  });
+});
+
+const addAverageOfLongitudeAggregation = () => {
```

Comment:
Looks like there are many local functions performing different actions in this test. Have you looked at https://github.com/metabase/metabase/blob/master/e2e/support/helpers/e2e-bi-basics-helpers.js whether existing helpers would help or would some of these new functions should live there?

Reply:
Now it looks much better, thanks!

f957882e95d3a5659c658c6237043e8f730f6251 + 7aa026bcb85f6deebbb2ec271629bae0d43c1125